app/src/main/java/io/gnosis/safe/ui/transaction/list/TransactionPagingRepository.kt

```diff
@@ -0,0 +1,26 @@
+package io.gnosis.safe.ui.transaction.list
+
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import io.gnosis.data.models.Transaction
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.data.repositories.TransactionRepository
+import kotlinx.coroutines.flow.Flow
+import pm.gnosis.model.Solidity
+
+class TransactionPagingRepository(
+    private val safeRepository: SafeRepository,
+    private val transactionRepository: TransactionRepository) {
+
+    fun getTransactionsStream(safe: Solidity.Address): Flow<PagingData<Transaction>> {
+        return Pager(
+            config = PagingConfig(pageSize = NETWORK_PAGE_SIZE, enablePlaceholders = false),
+            pagingSourceFactory = { TransactionPagingSource(safe, safeRepository, transactionRepository) }
+        ).flow
+    }
+
+    companion object {
+        private const val NETWORK_PAGE_SIZE = 50
```

Comment:
Is this required by the framework? I think in the case of our backend it's actually 20.

Reply:
we need some default value. 20 is small in my opinion. You will be initiating lots of requests and ui experience might suffer.

Reply:
Well, AFAIK the backend returns exactly 20 txs per page anyway, and we are not meant to fiddle with offsets and limit anyhow.

Reply:
there is no fiddling with offsets, but initial page size has to be known

Reply:
yeah, I am aware of that. But I was just letting you know that this value is known and it is `20`. If it doesn't need to match then it's all good :)